packages/vite/src/node/plugins/asset.ts

```diff
@@ -257,6 +257,11 @@ export async function urlToBuiltUrl(
   const file = url.startsWith('/')
     ? path.join(config.root, url)
     : path.join(path.dirname(importer), url)
+  const stat = await fsp.stat(cleanUrl(file))
+  if (stat.isDirectory()) {
+    return url
+  }
+
```

Comment:
@fi3ework why is this check needed for this PR?

Reply:
If the file is a dir, Vite will throw an error in build mode but works fine in dev mode. So we ignore the dir URL and preserve it as is in build mode.

Reply:
It's not related to this PR, should I split it into another PR?

Reply:
@fi3ework thanks, let's split this into another PR so we can discuss and maybe add an extra test to avoid regressions later. It would help to keep a clean history.

Reply:
Yeah, also because we use squash-merges, it can be references as a single commit then later in main branch history

Reply:
Done, I'll create another PR for dir error 😉